bccsp/pkcs11/cache.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package pkcs11
+
+import (
+	"crypto/ecdsa"
+	"sync"
+
+	"github.com/miekg/pkcs11"
+)
+
+type p11Key struct {
+	privH  pkcs11.ObjectHandle
+	pubH   pkcs11.ObjectHandle
+	pubKey *ecdsa.PublicKey
+}
+
+type keyCache struct {
+	mu      sync.RWMutex
+	p11Keys map[string]p11Key
+}
+
+func (k *keyCache) get(ski string) (p11Key, bool) {
```

Comment:
The map needs a comparable key but the function doesn't. Maybe key conversion moves here?

Reply:
I did think about doing the conversion in the get/set methods.  I'll go back and look, but I think I moved it out because there was a need for the conversation outside the cache.

Reply:
done